Snakefile

```diff
@@ -211,12 +289,14 @@ rule add_h5_clade:
     message: "Adding in a column for h5 clade numbering"
     input:
         metadata = "results/{subtype}/metadata.tsv",
-        clades_file = files.clades_file
+        clades_file = lambda w: resolve_config_path(files.clades_file, w)
```

Comment:
`lambda w: resolve_config_path(p, w)` might be better as `resolve_config_path(p)`, where `resolve_config_path()` returns an input function that takes wildcard values.  This allows for late-binding of wildcards, which can be useful, and it's also more ergonomic for rule authoring.

Or alternatively, `lambda w: resolve_config_path(expand(p, w)[0])`, e.g. totally separate/isolate `resolve_config_path()` from wildcard handling.  In this case, ihe index access could also be simplified away by making `resolve_config_path()` do something like:

```python
if isinstance(p, str):
    return _resolve_config_path(p)
else:
    return list(map(_resolve_config_path, p))
```

Reply:
Using a HoF like `param = resolve_config_path(p)` is really nice, I think i'll do that. Similarly for the `resolve_config_value` function in #104.

Reply:
PR has now shifted to using `param = resolve_config_path(p)`